Contributing.md

```diff
@@ -1,30 +1,472 @@
+Ideally, if someone reads the `git log` for your PR, they'll have a clear picture of why and how the changes were 
+introduced.
+
+## Code reviews
+
+Post mainnet, PRs require 3 or more positive reviews from previous contributors, including at least one positive review 
+from a contributor with write access to the repo.
+
+When reviewing PRs, here are some guidelines and suggestions to help maintain the safety and quality of code:
+
+* PULL the code and review and test it locally on your machine.
+* Do not spend more than 1 hour reviewing a PR at a time. 
+* Do not review more than 400 LOC in an hour.
+
+If either of these guidelines cannot be met, you may label the PR `CR-too_long` and politely ask the contributor to 
+revise their PR. Otherwise, take breaks and review the PR over multiple sessions or days.
+
+* Obviously, your main goal of the review is to 
+  * find bugs,
+  * identify edge cases that aren't handled properly,
+  * ensure that the stated goal of functions (as per the docstrings and/or RFC) matches the code as written.
+
+All **public** methods and functions must have a decent docstring.
+If docstrings are missing, and the purpose of a function or method is not clear, then ask the contributor to  
+provide it by tagging the PR with `CR-docs`.
+
+* Read the commit messages! They give the context for the change and why (according to the author) they are required.
+
+If there is insufficient justification in your view, or the git messages are too ["haaaands"y], then label the PR 
+`CR-insufficient_context` and ask the author to address this.
+
+If the PR tends to change things arbitrarily  because the "author likes it that way" (e.g. they change from _this_ 
+crate to _that_ crate, or names of things get changed seemingly arbitrarily), we should be wary of merging the PR 
+unless there are substantial performance, security, or readability reasons. This justification must be 
+provided in the commit messages. 
+If not, label the PR `CR-insufficient_justification`. If the author does not provide 
+additional justification AND the other reviewers agree with your assessment, then this is grounds for rejecting 
+and closing the PR.
+
+There are also some non-goals for a code review:
+* Do NOT make comments related to formatting or linting. If our linter doesn't care, neither should you. If the 
+  linter doesn't like something, the CI tasks will catch it and fail the PR.
+* You MAY make Clippy-like suggestions for making code cleaner, more Rust-y, or more readable. But be aware that we 
+  _do_ run Clippy as part of the CI process and require all errors to be resolved.
+* There should be decent test coverage, but 100% coverage is an ideal, not a requirement.
+
+If PR doesn’t have 100% coverage or breaks something non-critical, we may still merge it at our discretion as long as
+* Issues have been generated and triaged to cover the gaps.
+* The changes are behind a feature flag.
```

Comment:
Are the reviewers tasked to make sure we approach the ideal 100% test coverage? Are we looking at some tools to help us? For example, if a quantitative test coverage number decrease by introducing the PR, it is a good indication that we added untested code.